skorch/net.py

```diff
@@ -243,11 +236,33 @@ class NeuralNet:
       The complete (i.e. default and other), initialized callbacks, in
       a tuple with unique names.
 
+    modules_ : list of str
+      List of names of all modules that are torch modules. This list is
+      collected dynamically when the net is initialized. Typically, there is no
+      reason for a user to modify this list.
+
+    criteria_ : list of str
+      List of names of all criteria that are torch modules. This list is
+      collected dynamically when the net is initialized. Typically, there is no
+      reason for a user to modify this list.
```

Comment:
Is there any reason for a developer to directly interact with `criteria_`? Currently, I think its auto populated when one does `self.module2_  = ...`.